StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1119StatementMustNotUseUnnecessaryParenthesis.cs

```diff
@@ -60,7 +66,61 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(HandleParenthesizedExpressionn, SyntaxKind.ParenthesizedExpression);
+        }
+
+        private void HandleParenthesizedExpressionn(SyntaxNodeAnalysisContext context)
+        {
+            var node = context.Node as ParenthesizedExpressionSyntax;
+
+            if (node != null && node.Expression != null)
+            {
+                if (!(node.Expression is BinaryExpressionSyntax) && !(node.Expression is AssignmentExpressionSyntax)
+                    && !node.Expression.IsKind(SyntaxKind.CastExpression) && !node.Expression.IsKind(SyntaxKind.ConditionalExpression)
+                    && !(node.Expression is PrefixUnaryExpressionSyntax) && !(node.Expression is PostfixUnaryExpressionSyntax)
+                    && !node.Expression.IsKind(SyntaxKind.IsExpression) && !node.Expression.IsKind(SyntaxKind.SimpleLambdaExpression)
+                    && !node.Expression.IsKind(SyntaxKind.ParenthesizedLambdaExpression) && !node.IsKind(SyntaxKind.ConstructorDeclaration)
+                    && !node.Expression.IsKind(SyntaxKind.ArrayCreationExpression) && !node.Expression.IsKind(SyntaxKind.CoalesceExpression) 
+                    && !node.Expression.IsKind(SyntaxKind.QueryExpression))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, node.GetLocation()));
+                }
+                else
+                {
+                    if (!(node.Parent is ExpressionSyntax) || node.Parent.IsKind(SyntaxKind.VariableDeclaration)
```

Comment:
:question: Also, is a `VariableDeclaration` an `ExpressionSyntax`? If not I believe the test for `VariableDeclaration` is unnecessary.